Tasks/Common/packaging-common/nuget/NuGetToolGetter.ts

```diff
@@ -96,9 +99,33 @@ export async function getNuGet(versionSpec: string, checkLatest?: boolean, addNu
     let fullNuGetPath: string = path.join(toolPath, NUGET_EXE_FILENAME);
     taskLib.setVariable(NUGET_EXE_TOOL_PATH_ENV_VAR, fullNuGetPath);
 
+    // create a nuget posix script for nuget exe in non-windows agents
+    if (os.platform() !== "win32") {
+        generateNugetScript(toolPath, fullNuGetPath);
+    }
+    
     return fullNuGetPath;
 }
 
+function generateNugetScript(nugetToolPath: string, nugetExePath: string) {
+    var nugetScriptPath = path.join(nugetToolPath, "nuget");
+    taskLib.debug(`create nugetScriptPath ${nugetScriptPath}`);
+
+    fs.writeFile(
+        nugetScriptPath,
+        `#!/bin/sh\nmono ${nugetExePath} "$@"\n`,
+        (err) => {
+        if (err) {
+            taskLib.debug("Writing nuget script failed with error: " + err);
+        } else {
+            // give read and execute permissions to everyone
+            fs.chmodSync(nugetScriptPath, "555");
```

Comment:
Would it be a security issue for opening this for everybody on the machine?

Reply:
updated it to 500